### DISCUSSION TITLE
How to load inn_edsr_4x.pt?

### DISCUSSION THREAD

**Discussion by dtdo90** [TheStageAI/TorchIntegral#30] [Q&A]
Dear authors,

I have a question on how to use the state dict inn_edsr_4x.pt that was given in the google drive link. Thank you very much for spending time reading my question!

Based on edsr.py, we should convert the original 4xEDSR into an inn model and load this state dict, but I encounter errors which indicate **unexpected key** in state dict.  

The code I used is as in the file edsr.py:

model = EdsrModel.from_pretrained("eugenesiow/edsr", scale=4)

conv = model.tail[0][0]
model.tail[0][0] = torch.nn.Conv2d(conv.in_channels,conv.in_channels, conv.kernel_size,conv.stride,conv.padding,bias=False,)

model.tail[0][1] = torch.nn.Upsample(scale_factor=2)

conv = model.tail[0][2]
model.tail[0][2] = torch.nn.Conv2d(conv.in_channels, conv.in_channels, conv.kernel_size, conv.stride, conv.padding, bias=False,)

model.tail[0][3] = torch.nn.Upsample(scale_factor=2)

continuous_dims = standard_continuous_dims(model)
discrete_dims = { "sub_mean.weight": [0, 1], "sub_mean.bias": [0], "add_mean.weight": [0, 1], "add_mean.bias": [0], "head.0.weight": [1],  "tail.0.0.weight": [0], "tail.0.2.weight": [0, 1], "tail.1.weight": [0, 1],}

example_input = (1, 3, 32, 32)
model = inn.IntegralWrapper(
    init_from_discrete=True,
    permutation_config={"class": NOptOutFiltersPermutation},
)(model, example_input, continuous_dims, discrete_dims)

model = model.transform_to_discrete()

state_dict = {
    key.replace("module.", ""): value
    for key, value in torch.load("inn_edsr_4x.pt").items()}

model.load_state_dict(state_dict)

<img width="1093" alt="Screenshot 2023-11-09 at 2 11 50 PM" src="https://github.com/TheStageAI/TorchIntegral/assets/145887647/2d6e7b6c-e10b-4070-88bf-1b6902160a19">

**Comment by b1n0** [comment]
Hi! The checkpoint provided in the google drive is an integral model, so you should `load_state_dict` before calling `transform_to_discrete()` method.

**Reply by dtdo90** [reply]
I have tried that, but there are still **missing keys** and **size mismatch** errors

<img width="1200" alt="Screenshot 2023-11-10 at 8 40 39 AM" src="https://github.com/TheStageAI/TorchIntegral/assets/145887647/705a63cf-0e0c-4768-9947-2fa0b902be0b">

<img width="1201" alt="Screenshot 2023-11-10 at 8 40 53 AM" src="https://github.com/TheStageAI/TorchIntegral/assets/145887647/1689248e-9c9a-48eb-90ff-c46cf1af815e">
